super_dpf.py

```diff
@@ -1,127 +1,316 @@
+
+
+class LogManager(object):
+    LOG_LOCATION = os.path.dirname(os.path.realpath(__file__))
+    LOGGING = {
+        'version': 1,
+        'formatters': {
+            'standard': {
+                'format': "[%(asctime)s] %(levelname)s [%(name)s:%(lineno)s] %(message)s",
+                'datefmt': "%d/%b/%Y %H:%M:%S"
+            },
+            'verbose': {
+                'format': 'dpf[%(process)d]: %(levelname)s %(name)s[%(module)s] %(message)s'
+            },
+        },
+        'handlers': {
+            'null': {
+                'level': 'DEBUG',
+                'class': 'logging.NullHandler',
+            },
+            # 'mail_admins': {
+            #     'level': 'ERROR',
+            #     'class': 'logging.handlers.AdminEmailHandler'
+            # },
+            'logfile': {
+                'level': 'INFO',
+                'class': 'logging.handlers.RotatingFileHandler',
+                'filename': '{}/sdpf.log'.format(LOG_LOCATION),
+                'maxBytes': 50000000,
+                'backupCount': 2,
+                'formatter': 'standard',
+            },
+            'console': {
+                'level': 'DEBUG',
+                'class': 'logging.StreamHandler',
+                'formatter': 'standard'
+            },
+            'syslog': {
+                'level': 'INFO',
+                'class': 'logging.handlers.SysLogHandler',
+                'formatter': 'verbose',
+                'address': '/dev/log',
+                'facility': 'local2',
+            }
+        },
+        'loggers': {
+            'django': {
+                'handlers': ['console', 'syslog'],
+                'propagate': True,
+                'level': 'WARN',
+            },
+            'urllib3': {
+                'handlers': ['console', 'logfile', 'syslog'],
+                'propagate': True,
+                'level': 'ERROR',
+            },
+            '': {
+                'handlers': ['console', 'logfile', 'syslog'],
+                'propagate': True,
+                'level': 'DEBUG',
+            },
+        }
+    }
 
+    dictConfig(LOGGING)
+
+
+class BaseDPF(DPFConfigurator, LogManager):
     @property
     def settings(self):
         try:
             with open(self.PATHS['config'], 'r') as config_file:
                 config = json.load(config_file)
             return config
-        except IOError:
-            print 'There was an error loading the settings file'
-            return
+        except IOError as e:
+            logging.error(e)
+
+    def _save_settings(self, **kwargs):
+        config = self._walk_json_tree(self.settings, kwargs)
+        with open(self.PATHS['config'], 'w') as config_file:
+            json.dump(config, config_file)
 
+
+class AWSController(BaseDPF):
+    @property
+    def aws_dir(self):
+        return '{}/{}'.format(self.PATHS['photos'], 'aws')
+    
     @property
-    def is_first_run(self):
-        return not self.settings
+    def aws_configured(self):
```

Comment:
Providing service specific prefixes such as 'aws_' makes it so that other code has to know of class specific calling behavior.  The interface should be defined and all of the interface's required functions whose names are always the same should be implemented once for each service class.